src/shared/lib/services/Logger.ts

```diff
@@ -0,0 +1,286 @@
+import { FileSystem } from 'react-native-file-access';
+import { FileLogger } from 'react-native-file-logger';
+import RNFetchBlob from 'rn-fetch-blob';
+
+import { FileService } from '@app/shared/api';
+
+import { IS_ANDROID } from '../constants';
+import { IMutex, Mutex, isAppOnline } from '../utils';
+
+const isAndroid = IS_ANDROID;
+
+const isIos = !isAndroid;
+
+interface ILogger {
+  log: (message: string) => void;
+  info: (message: string) => void;
+  warn: (message: string) => void;
+  error: (message: string) => void;
+  configure: () => void;
+  send: () => Promise<boolean>;
+  cancelSending: () => void;
+  clearAllLogFiles: () => Promise<void>;
+}
+
+type NamePath = {
+  fileName: string;
+  filePath: string;
+};
+
+type NamePathSize = {
+  size: number;
+} & NamePath;
+
+type FileExists = {
+  exists: boolean;
+} & NamePathSize;
+
+class Logger implements ILogger {
+  private mutex: IMutex;
+
+  private cancellationRequested: boolean;
+
+  constructor() {
+    this.mutex = Mutex();
+    this.cancellationRequested = false;
```

Comment:
There should be another way to cancel stuff. Please try using AbortController

Reply:
no need, let's just stop uploading the current file?